tests/BunqSdkTestBase.php

```diff
@@ -3,61 +3,246 @@
+        );
+
+        $this->secondMonetaryAccountBank = MonetaryAccountBank::get($createdId->getValue())->getValue();
+    }
+
+    /**
+     * @return Pointer
+     *
+     * @throws BunqException
+     */
+    protected function getSecondMonetaryAccountAlias(): Pointer
+    {
+        $allAlias = $this->secondMonetaryAccountBank->getAlias();
+
+        foreach ($allAlias as $alias) {
+            if ($alias->getType() === self::POINTER_TYPE_IBAN) {
+                return $alias;
+            }
         }
 
-        $apiContext->save(static::FILENAME_CONTEXT_CONFIG);
+        throw new BunqException(self::ERROR_COULD_NOT_DETERMINE_IBAN_POINTER);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getAttachmentFilePath(): string
+    {
+        return self::ATTACHMENT_PATH_IN;
+    }
 
-        BunqContext::loadApiContext($apiContext);
+    /**
+     * @return Pointer
+     *
+     * @throws BunqException
+     */
+    protected function getUserAlias(): Pointer
+    {
+        if (BunqContext::getUserContext()->isOnlyUserPersonSet()) {
+            return BunqContext::getUserContext()->getUserPerson()->getAlias()[self::INDEX_FIRST];
+        } elseif (BunqContext::getUserContext()->isOnlyUserCompanySet()) {
+            return BunqContext::getUserContext()->getUserCompany()->getAlias()[self::INDEX_FIRST];
+        } else {
+            throw new BunqException(self::ERROR_COULD_NOT_DETERMINE_USER_ALIAS);
+        }
     }
 
     /**
-     * @return ApiContext
+     * @return Pointer
      */
-    protected static function createApiContext(): ApiContext
+    protected function getUserBravoPointer(): Pointer
     {
-        return ApiContext::create(
-            BunqEnumApiEnvironmentType::SANDBOX(),
-            Config::getApiKey(),
-            static::DEVICE_DESCRIPTION,
-            Config::getPermittedIps()
+        return new Pointer(
+            self::POINTER_TYPE_EMAIL,
+            self::EMAIL_BRAVO
         );
     }
+
+    /**
+     * @return int
+     */
+    protected function getSecondMonetaryAccountId(): int
+    {
+        return $this->secondMonetaryAccountBank->getId();
+    }
+
+    /**
+     * @return int
+     */
+    protected function getCashRegisterId(): int
+    {
+        if (is_null($this->cashRegister)) {
+            $this->setCashRegister();
+        }
+
+        return $this->cashRegister->getId();
+    }
+
+    /**
+     * @return bool
+     */
+    protected function doesAccountHaveEnoughMoney(): bool
+    {
+        $balance = floatval(BunqContext::getUserContext()->getPrimaryMonetaryAccount()->getBalance()->getValue());
+
+        return $balance > self::MONETARY_ACCOUNT_BALANCE_THRESHOLD;
+    }
+
+    /**
+     * @return bool
+     */
+    protected function assertTestShouldBeSkippedDueToInsufficientBalance(): bool
+    {
+        if (!$this->doesAccountHaveEnoughMoney()) {
+            static::markTestSkipped(self::WARMING_TEST_SKIPPED_DUE_TO_INSUFFICIENT_BALANCE);
+        }
```

Comment:
Explicit else

Reply:
In open-source project explicit else is not needed. Only in bunq internal projects.